### PR TITLE
Move Tril's implementation of enable_if to the OMR core utilities

### PR DESCRIPTION
**Pull request by Leonardo2718**
This change set adds an `EnableIf` utility to the OMR core utilities as a replacement for the custom implementation in Tril

Closes: #3320

FYI @rwy0717 @youngar @dnakamura

### CODE REVIEW

**Comment by charliegracie** [comment]
@rwy0717 can you review this please?

**Comment by rwy0717** [comment]
Looks good, thanks @Leonardo2718 !

File: fvtest/coretest/TestTypeTraits.cpp [hunk 1]
```diff
@@ -113,4 +113,23 @@ TEST(TestTypeTraits, IsVoid)
 	EXPECT_FALSE((IsVoid<void*>::VALUE));
 }
```
Comment (0xdaryl) [inline fvtest/coretest/TestTypeTraits.cpp]: This file requires a proper copyright header.  Can you please add one in a separate commit?

File: fvtest/tril/tril/ast.hpp [hunk 2]
```diff
@@ -23,7 +23,8 @@
 #define AST_HPP
```
Comment (0xdaryl) [inline fvtest/tril/tril/ast.hpp]: Please update the copyright header in this file to 2018.

**Comment by Leonardo2718** [comment]
@0xdaryl Added/updated copy-right headers as requested and also fixed some formatting inconsistencies.

@genie-omr build all

**Comment by Leonardo2718** [comment]
@genie-omr build all

**Pull request merged**